draco/core/task.py

```diff
@@ -364,6 +369,10 @@ def finish(self):
 
     def _process_output(self, output, input_tag=None):
 
+        # when output is None, there is nothing to write or check
+        if output is None:
+            return None
```

Comment:
I think something more dramatic should happen here. This routine should only ever be called with a valid memh5 container, and so it's probably more appropriate to raise a specific exception than silently return None. Probably this test should be something more specific like `not isinstance(output, memh5.MemDiskGroup)` (I am not entirely sure that's the appropriate class to test for).

The same is true of `_nan_process_output`.

Reply:
>  This routine should only ever be called with a valid memh5 container

This makes sense to me

>  so it's probably more appropriate to raise a specific exception than silently return None.

This makes sense to me if you get a non-valid memh5 container. Though, then if something like SiderealGrouper returns an `output` of `None`, that results in an `Exception`? Is this the right thing to happen?

Reply:
- [ ] I am also going to add a small docstring to `__process_output` as part of this PR.

Reply:
I used `memh5.MemDiskGroup` and `memh5.MemGroup`, but got this exception with the tests:

```
        if not isinstance(output, memh5.MemDiskGroup) or not isinstance(
            output, memh5.MemGroup
        ):
            raise pipeline.PipelineRuntimeError(
>               f"Task must output a valid memh5 container; given {type(output)}"
            )
E           caput.pipeline.PipelineRuntimeError: Task must output a valid memh5 container; given <class 'draco.core.containers.SiderealStream'>
```

Is the test realistically depicting the standard pipeline?

Reply:
> Is the test realistically depicting the standard pipeline by returning a `SiderealStream` container?

Yes.

The bug is in the boolean logic. You almost certainly want `not (isinstance(output, memh5.MemDiskGroup) or isinstance(output, memh5.MemGroup)` or the more succinct `not isinstance(output, (memh5.MemDiskgroup, memh5.MemGroup))`.

What you've written is equivalent to `not (isinstance(output, MemDiskgroup) and isinstance(output, MemGroup))`

Reply:
Man, you are totally right, what was I doing, I was thinking in English.

Reply:
TBH you probably don't want to check for MemGroup at least within the branch that deals with saving. Ultimately it calls `output.save(...)` (within `BasicContMixin.write_output`) and that *only* works on subclasses of `MemDiskGroup` , but will fail if given a `MemGroup`.